File: src/main/java/nl/crosscode/x509certbuddy/actions/CopyBase64Action.java
package nl.crosscode.x509certbuddy.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import nl.crosscode.x509certbuddy.ui.Exporters;
import nl.crosscode.x509certbuddy.ui.x509CertAssistant;
import nl.crosscode.x509certbuddy.x509CertAssistantFactory;
import org.jetbrains.annotations.NotNull;

public class CopyBase64Action extends AnAction {
    private final Exporters exporters;
    public CopyBase64Action(Exporters exporters) {
        this.exporters = exporters;
        this.getTemplatePresentation().setText("Copy Base64");
    }

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        exporters.copyBase64(null);
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        x509CertAssistant assistant = x509CertAssistantFactory.getX509CertAssistant(e.getProject());
        e.getPresentation().setEnabled(assistant!=null?assistant.hasCertSelected():false);
        super.update(e);
    }
}

File: src/main/kotlin/nl/crosscode/x509certbuddy/actions/CopyBase64Action.kt
package nl.crosscode.x509certbuddy.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import nl.crosscode.x509certbuddy.ui.Exporters
import nl.crosscode.x509certbuddy.X509CertAssistantFactory

class CopyBase64Action(private val exporters: Exporters) : AnAction() {
    init {
        this.templatePresentation.text = "Copy Base64"
    }

    override fun actionPerformed(e: AnActionEvent) {
        exporters.copyBase64()
    }

    override fun update(e: AnActionEvent) {
        if (e.project == null) {
            e.presentation.isEnabled = false
            return
        }
        val assistant = X509CertAssistantFactory.getInstance(e.project!!)
        e.presentation.isEnabled = assistant?.hasCertSelected() ?: false
        super.update(e)
    }

    override fun getActionUpdateThread(): ActionUpdateThread {
        return ActionUpdateThread.BGT
    }
}
